### ISSUE TITLE
Failure to detect read-only declare statements being written to

### ISSUE THREAD

**Issue by rbairwell** [koalaman/shellcheck#2525]
#### For bugs
- Rule Id (if any, e.g. SC1000):  None
- My shellcheck version (`shellcheck --version` or 'online'): `online` and ` 0.7.1`
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

Example 1:
```sh
#!/bin/bash
declare -r script_name="${0}"
script_name="test"
echo "${script_name}"
```

Example 2:

```sh
#!/bin/bash
declare -r script_name="${0}"
declare -r script_name="test"
echo "${script_name}"
```

#### Here's what shellcheck currently says:

```
No issues detected!
```

#### Here's what I wanted or expected to see:

Example 1:
A warning along the lines of `Cannot set contents of script_name variable at line 3 : originally declare'd readonly on line 2`

Example 2:
A warning along the lines of `Readonly variable "script_name" redeclared at line 3 : originally declare'd on line 2`  (bash reports `declare: script_name: readonly variable`)
